pkg/repo/chartrepo_test.go

```diff
@@ -319,3 +324,50 @@ func TestResolveReferenceURL(t *testing.T) {
 		t.Errorf("%s contains query string from base URL when it shouldn't", chartURL)
 	}
 }
+
+func TestChartRepository_DownloadIndexFile(t *testing.T) {
+	t.Run("should not decode the path in the repo url", func(t *testing.T) {
+		ctrl := gomock.NewController(t)
+		defer ctrl.Finish()
+
+		r, err := NewChartRepository(&Entry{
+			Name:  "testrepo",
+			URL:   "http://example-charts.com/some%2Fpath/test",
+			Cache: "/repository/cache/testrepo-index.yaml",
+		}, getter.All(environment.EnvSettings{}))
+		if err != nil {
+			t.Errorf("Problem creating chart repository from %s: %v", "testrepo", err)
+		}
+
+		fs := afero.NewMemMapFs()
+		r.Fs = fs
+
+		err = fs.MkdirAll("/repository/cache/", 0644)
+		assert.NoError(t, err, "problem creating test repository cache")
+
+		mockGetter := mocks.NewMockGetter(ctrl)
+		r.Client = mockGetter
+		indexYamlContent := `apiVersion: v1
```

Comment:
While I appreciate the testing the style of this code and the use of `github.com/spf13/afero` is different from the way the rest of the Helm code base handles things. The typical situation is to use a temp directory (search for `ensure.TempDir` to find examples) and the repo package has a method for creating and working with a test server.

Would it be possible to use the other setups found in Helm for consistency? Or, is there a reason to break from that.

Reply:
I did notice the temp dir style testing recently in some places. Will try to do the same 👍 and remove tmp for once test is over

I do want to mention some pros for afero though - it makes testing very easy - no need to delete files, you can use in-memory fs and write to anywhere and not just tmp directories

Reply:
I would suggest posing that as a follow-up PR to enhance all unit tests. Same with golang/mock... Up to now, we've been using `net/http/httptest` for most tests with chart repositories. For example: https://github.com/helm/helm/blob/7cad59091a9451b2aa4f95aa882ea27e6b195f98/pkg/repo/chartrepo_test.go#L241

Reply:
Sure 👍 I will raise issue to discuss the methods / strategies to test and then raise separate PRs for them